docs/modules/ROOT/pages/deploy_related/integration_keystone.adoc

```diff
@@ -0,0 +1,98 @@
+= Integration with OpenStack Keystone
+v0.1, 2020-05-28
+:toc: right
+:imagesdir: ../../images
+
+NOTE: This document is under development
+
+== Overview
+
+Alcor API currently requires a project id in the API Url.Neutron 2.0 APIs, however, don't need to provide a project id as
+it generates the project id based on the Authentication token in the X-Auth-Token request header, when Keystone is enabled.
+
+Reference: https://docs.openstack.org/api-ref/network/v2/
+"The Networking API v2.0 uses the OpenStack Identity service as the default authentication service.
+When Keystone is enabled, users that submit requests to the OpenStack Networking service must provide an authentication
+token in X-Auth-Token request header. You obtain the token by authenticating to the Keystone endpoint.
+
+When Keystone is enabled, the project_id attribute is not required in create requests
+because the project ID is derived from the authentication token."
+
+[#system-requirements]
+== Integration Requirements
+
+. Identify Neutron's algorithm that generates project id based on X-Auth-Token.
+. Make changes to API Gateway to support Authentication & Authorization with KeyStone, and support same way how project id is generated based on X-Auth-Token in API Gateway.
+. Integration testing with KeyStone.
+
+== Review of OpenStack Workflow
+https://docs.openstack.org/api-ref/identity/v3/index.html#check-token[Openstack keystone apis]
+
+In openstack neutron project, it use keystonemiddle to verify request token. .
```

Comment:
@Gzure Are we following similar steps as keystonemiddle?

Reply:
Most steps are similar to keystonemiddle。